packages/docs/course-guide/README.md

```diff
@@ -35,3 +35,155 @@ __User Journey Statechart__
+
+## Part 1: Program Choosing
+
+### MyProgramsList
+
+The list of programs the user has added. 
+
+```js
+MyProgramsList{
+    data:{
+        programs: //array of programs the user has added
+    },
+    methods:{
+        deleteProgram(program) // triggered when pressing delete on a program
+    },
+    components:{
+        ChosenProgramTile //each tile is a program in the list and is clickable
+    }
+}
+```
+
+### ChosenProgramTile
+
+The component for the programs listed in My_programs_List
+
+```js
+ChosenProgramTile{
+    data: {
+        programName: String //The name of the program this tile represents
+        subject //which subject this program belongs in
+    },
+    methods: {
+        loadSubject(subject), /*displays the programs in the same subject as 
+        selected_program in the "programCardsPanel",*/
+        delete() //remove this program when the "x" button is clicked
+    },
+    components: {
+        deleteButton
+    }
+}
+```
+
+### Subjects_List
+
+The Subjects_List is a list of all the subject names offered at UTM. The user can browse through it and click a subject in this list to view all the programs of that subject.
+
+```js
+Subjects_List{
+    data:{
+        selectedSubject, //The selected subject needs to be highlighted
+        searchMatchingSubjects: subject[] /*The subjects that have a program that 
+                                            matches the search query*/
+        showingGroups //array of groups that still shows in this list
+    },
+    components:{
+        SubjectTiles: //an array of subject names that can be clicked
+    }
+}
+```
+
+### Subject_Tile
+
+The component for the subjects in Subjects_List
+
+```js
+Subject_Tile{
+    data: {
+        subject, //the subject this tile represents
+        highlighted: boolean
+    },
+    methods: {
+        loadSubject(subject) /*displays the programs in the same subject as 
+        selected_program in the "programCardsPanel",*/
+    }
+}
+```
+
+### programCardsPanel
+
+The component that displays the detailed information of each program in a subject and allows the user to add a program to My_Programs_List
+
+```js
+programCardsPanel {
+    data: {
+        subject //The subject 
+    },
+    methods:{
+        setSubject(subject) //Save the specified subject into data.subject to load programCards
+    },
+    components:{
+        ProgramCards: //each individual program's information 
+    }
+}
+```
+
+### ProgramCard
+
+The ProgramCard displays all the information about a specific program, and a button to add it
+
+```js
+ProgramCard {
+    data:{
+        program //The program this card represents
+    },
+    methods: {
+        /*The required courses string for a certain year contains course codes
+        that refer to a course, we need a regex to recognize the course code pattern and transform
+        each of them into a CourseLink component that can be clicked to pop up a course info frame.*/
+    },
+    sub_components:{
```

Comment:
Please use the Vue syntax as follows: 
<img width="255" alt="image" src="https://user-images.githubusercontent.com/19757203/75312319-81a0f680-5827-11ea-875d-e4aabb90ba46.png">